File: net-examples/Aspose.Font.Examples/ConvertFont/ConvertCff.cs
using Aspose.Font.Sources;
using System;
using System.IO;

namespace Aspose.Font.Examples.ConvertFont
{
    class ConvertCff : BaseExamples
    {
        public ConvertCff(string folderName) : base(folderName) { }

        public override void Run()
        {
            Console.WriteLine("Run Convert CFF to TTF/WOFF/WOFF2 examples");

            CffToTtfExample1();
            CffToTtfExample2();
            CffToWoffExample();
            CffToWoff2Example();
        }

        

        private void CffToTtfExample1()
        {
            Console.WriteLine("CFF to TTF ex: 1");

            //ExampleStart: 1            
            // Open cff font
            string fontPath = Path.Combine(DataDir, "CenturyGothic.cff");
            FontDefinition fontDefinition = new FontDefinition(FontType.CFF, new FontFileDefinition("cff", new FileSystemStreamSource(fontPath)));
            Font font = Font.Open(fontDefinition);

            // Ttf output settings
            string outPath = Path.Combine(OutputDir, "CffToTtf_out1.ttf");
            FileStream outStream = File.Create(outPath);

            // Convert cff to ttf
            font.SaveToFormat(outStream, FontSavingFormats.TTF);
            
            //ExampleEnd: 1
        }

        private void CffToTtfExample2()
        {
            Console.WriteLine("CFF to TTF ex: 2");

            //ExampleStart: 2
            
            // Open cff font
            string fontPath = Path.Combine(DataDir, "CenturyGothic.cff");
            FontDefinition fontDefinition = new FontDefinition(FontType.CFF, new FontFileDefinition("cff", new FileSystemStreamSource(fontPath)));
            Font font = Font.Open(fontDefinition);

            // Convert cff to ttf
            Font ttfFont = font.Convert(FontType.TTF) as Aspose.Font.Ttf.TtfFont;
            ttfFont.Save(Path.Combine(OutputDir, "CffToTtf_out2.ttf"));
            
            //ExampleEnd: 2
        }

        private void CffToWoffExample()
        {
            Console.WriteLine("CFF to WOFF ex: 3");

            //ExampleStart: 3
            
            // Open cff font
            byte[] fontMemoryData = File.ReadAllBytes(Path.Combine(DataDir, "CenturyGothic.cff"));
            FontDefinition fontDefinition = new FontDefinition(FontType.CFF, new FontFileDefinition("cff", new ByteContentStreamSource(fontMemoryData)));
            Font font = Font.Open(fontDefinition);

            // Woff output settings
            string outPath = Path.Combine(OutputDir, "CffToWoff_out3.woff");
            FileStream outStream = File.Create(outPath);

            // Convert cff to woff
            font.SaveToFormat(outStream, FontSavingFormats.WOFF);
            
            //ExampleEnd: 3
        }

        private void CffToWoff2Example()
        {
            Console.WriteLine("CFF to WOFF2 ex: 4");

            //ExampleStart: 4
            
            // Open cff font
            byte[] fontMemoryData = File.ReadAllBytes(Path.Combine(DataDir, "CenturyGothic.cff"));
            FontDefinition fontDefinition = new FontDefinition(FontType.CFF, new FontFileDefinition("cff", new ByteContentStreamSource(fontMemoryData)));
            Font font = Font.Open(fontDefinition);

            // Woff2 output settings
            string outPath = Path.Combine(OutputDir, "CffToWoff2_out4.woff2");
            FileStream outStream = File.Create(outPath);

            // Convert cff to woff2
            font.SaveToFormat(outStream, FontSavingFormats.WOFF2);
            
            //ExampleEnd: 4
        }
    }
}

File: cpp-examples/Aspose.Font.Examples.CPP/source/ConvertFont/ConvertCff.cpp
#include "aspose_pch.h"
#include "ConvertCff.h"

#include <system/io/path.h>
#include <system/io/file_stream.h>
#include <system/io/file.h>
#include <system/exceptions.h>
#include <system/console.h>
#include <system/array.h>
#include <Aspose.Font.Cpp/src/Ttf/TtfFont.h>
#include <Aspose.Font.Cpp/src/Sources/FontFileDefinition.h>
#include <Aspose.Font.Cpp/src/Sources/FontDefinition.h>
#include <Aspose.Font.Cpp/src/Sources/FileSystemStreamSource.h>
#include <Aspose.Font.Cpp/src/Sources/ByteContentStreamSource.h>
#include <Aspose.Font.Cpp/src/FontType.h>
#include <Aspose.Font.Cpp/src/FontSavingFormats.h>
#include <Aspose.Font.Cpp/src/Font.h>
#include <cstdint>


using namespace Aspose::Font::Sources;
namespace Aspose {

namespace Font {

namespace Examples {

namespace ConvertFont {

RTTI_INFO_IMPL_HASH(2208113303u, ::Aspose::Font::Examples::ConvertFont::ConvertCff, ThisTypeBaseTypesInfo);

ConvertCff::ConvertCff(System::String folderName) : Aspose::Font::Examples::BaseExamples(folderName)
{
}

void ConvertCff::Run()
{
    System::Console::WriteLine(u"Run Convert CFF to TTF/WOFF/WOFF2 examples");
    
    CffToTtfExample1();
    CffToTtfExample2();
    CffToWoffExample();
    CffToWoffExample2();
}

void ConvertCff::CffToTtfExample1()
{
    System::Console::WriteLine(u"CFF to TTF ex: 1");
    
    //ExampleStart: 1            
    // Open cff font
    System::String fontPath = System::IO::Path::Combine(get_DataDir(), u"CenturyGothic.cff");
    System::SharedPtr<FontDefinition> fontDefinition = System::MakeObject<FontDefinition>(Aspose::Font::FontType::CFF, System::MakeObject<FontFileDefinition>(u"cff", System::MakeObject<FileSystemStreamSource>(fontPath)));
    System::SharedPtr<Aspose::Font::Font> font = Aspose::Font::Font::Open(fontDefinition);
    
    // Ttf output settings
    System::String outPath = System::IO::Path::Combine(get_OutputDir(), u"CffToTtf_out1.ttf");
    System::SharedPtr<System::IO::FileStream> outStream = System::IO::File::Create(outPath);
    
    // Convert cff to ttf
    font->SaveToFormat(outStream, Aspose::Font::FontSavingFormats::TTF);
    
    //ExampleEnd: 1
}

void ConvertCff::CffToTtfExample2()
{
    System::Console::WriteLine(u"CFF to TTF ex: 2");
    
    //ExampleStart: 2
    
    // Open cff font
    System::String fontPath = System::IO::Path::Combine(get_DataDir(), u"CenturyGothic.cff");
    System::SharedPtr<FontDefinition> fontDefinition = System::MakeObject<FontDefinition>(Aspose::Font::FontType::CFF, System::MakeObject<FontFileDefinition>(u"cff", System::MakeObject<FileSystemStreamSource>(fontPath)));
    System::SharedPtr<Aspose::Font::Font> font = Aspose::Font::Font::Open(fontDefinition);
    
    // Convert cff to ttf
    System::SharedPtr<Aspose::Font::Font> ttfFont = System::Cast_noexcept<Aspose::Font::Ttf::TtfFont>(font->Convert(Aspose::Font::FontType::TTF));
    ttfFont->Save(System::IO::Path::Combine(get_OutputDir(), u"CffToTtf_out2.ttf"));
    
    //ExampleEnd: 2
}

void ConvertCff::CffToWoffExample()
{
    System::Console::WriteLine(u"CFF to WOFF ex: 3");
    
    //ExampleStart: 3
    
    // Open cff font
    System::ArrayPtr<uint8_t> fontMemoryData = System::IO::File::ReadAllBytes(System::IO::Path::Combine(get_DataDir(), u"CenturyGothic.cff"));
    System::SharedPtr<FontDefinition> fontDefinition = System::MakeObject<FontDefinition>(Aspose::Font::FontType::CFF, System::MakeObject<FontFileDefinition>(u"cff", System::MakeObject<ByteContentStreamSource>(fontMemoryData)));
    System::SharedPtr<Aspose::Font::Font> font = Aspose::Font::Font::Open(fontDefinition);
    
    // Woff output settings
    System::String outPath = System::IO::Path::Combine(get_OutputDir(), u"CffToWoff_out3.woff");
    System::SharedPtr<System::IO::FileStream> outStream = System::IO::File::Create(outPath);
    
    // Convert cff to woff
    font->SaveToFormat(outStream, Aspose::Font::FontSavingFormats::WOFF);
    
    //ExampleEnd: 3
}

void ConvertCff::CffToWoffExample2()
{
    System::Console::WriteLine(u"CFF to WOFF2 ex: 4");
    
    //ExampleStart: 4
    
    // Open cff font
    System::ArrayPtr<uint8_t> fontMemoryData = System::IO::File::ReadAllBytes(System::IO::Path::Combine(get_DataDir(), u"CenturyGothic.cff"));
    System::SharedPtr<FontDefinition> fontDefinition = System::MakeObject<FontDefinition>(Aspose::Font::FontType::CFF, System::MakeObject<FontFileDefinition>(u"cff", System::MakeObject<ByteContentStreamSource>(fontMemoryData)));
    System::SharedPtr<Aspose::Font::Font> font = Aspose::Font::Font::Open(fontDefinition);
    
    // Woff2 output settings
    System::String outPath = System::IO::Path::Combine(get_OutputDir(), u"CffToWoff2_out4.woff2");
    System::SharedPtr<System::IO::FileStream> outStream = System::IO::File::Create(outPath);
    
    // Convert cff to woff2
    font->SaveToFormat(outStream, Aspose::Font::FontSavingFormats::WOFF2);
    
    //ExampleEnd: 4
}

} // namespace ConvertFont
} // namespace Examples
} // namespace Font
} // namespace Aspose
